### PR TITLE
Support allow_duplicates on append processor

### PR DESCRIPTION
**Pull request by stevejgordon**
Related to https://github.com/elastic/elasticsearch/pull/61916
Contributes to #5198

### CODE REVIEW

File: tests/Tests/Ingest/ProcessorAssertions.cs [hunk 1]
```diff
@@ -59,19 +60,41 @@ public static IPromise<IList<IProcessor>> Fluent(ProcessorsDescriptor d)
...
+			public override IProcessor Initializer
 			{
-				Field = "state",
-				Value = new object[] { StateOfBeing.Stable, StateOfBeing.VeryActive }
-			};
+				get
+				{
+					var ap = new AppendProcessor { Field = "state", Value = new object[] { StateOfBeing.Stable, StateOfBeing.VeryActive } };
+
+					if (TestClient.Configuration.InRange(">=7.11.0"))
```
Comment (Mpdreamz) [inline tests/Tests/Ingest/ProcessorAssertions.cs:80]: Better to add a `SkipVersion` on the `Append` class and include the additional property. 

This way on master we can clean up the `SkipVersion` easier when we go over all the unnecessary skips.

**Review by Mpdreamz — CHANGES_REQUESTED** [review]
I think its okay to simplify the `Append` cases to only run on `7.11.0` or include a `AppendWithAllowDuplicates` additional test class. 

Easier to track then dynamically built json. 


Also failing on the following `CodeStandards` tests:

```
Tests.CodeStandards.Descriptors.DescriptorMethodsTakingSingleValueTypeShouldBeNullable [FAIL]
  Failed Tests.CodeStandards.Descriptors.DescriptorMethodsTakingSingleValueTypeShouldBeNullable [166 ms]
  Error Message:
   Expected breakingDescriptors to be empty, but found {"allowDuplicates on method AllowDuplicates of Nest.AppendProcessorDescriptor`1 is not nullable"}.
  Stack Trace:
```

Descriptors taking value types should be nullable so that they be effectively reset.

File: tests/Tests/Ingest/ProcessorAssertions.cs (hunk 1 above)
Comment (stevejgordon) [reply tests/Tests/Ingest/ProcessorAssertions.cs]: I've done as suggested in the comment and added a separate test which covers allow duplicates on 7.11+

**Comment by stevejgordon** [comment]
@Mpdreamz Ready for a re-review please

**Review by Mpdreamz — APPROVED** [review]
LGTM !

**Pull request merged**